checker/src/main/java/org/checkerframework/checker/nullness/NullnessTransfer.java

```diff
@@ -264,21 +264,25 @@ protected NullnessValue finishValue(NullnessValue value, NullnessStore store) {
 
             if (keyForTypeFactory.isKeyForMap(mapName, methodArgs.get(0))
                     && !hasNullableValueType((AnnotatedDeclaredType) receiverType)) {
-                makeNonNull(result, n);
-
-                NullnessValue oldResultValue = result.getResultValue();
-                NullnessValue refinedResultValue =
-                        analysis.createSingleAnnotationValue(
-                                NONNULL, oldResultValue.getUnderlyingType());
-                NullnessValue newResultValue =
-                        refinedResultValue.mostSpecific(oldResultValue, null);
-                result.setResultValue(newResultValue);
+                refineToNonNull(result, n);
             }
         }
 
         return result;
     }
 
+    /** Refine the given result to @NonNull. */
+    void refineToNonNull(
```

Comment:
This method is also refining the MethodInvocationNode `n` to non-null, but that's not mentioned in the Javadoc.  I would move the call to `makeNonNull(result, n)` out of this method.  Then the parameter `n` isn't needed either.